code/modules/projectiles/guns/chaos_bolt.dm

```diff
@@ -0,0 +1,348 @@
+	var/mob/living/carbon/human/H = target
+	switch(chaos_effect)
+		if("ded")
+			H.visible_message("<span class='chaosverybad'>[H] drops dead!</span>", "<span class='chaosverybad'>Game over!</span>")
+			H.death()
+		if("heart deleted")
+			H.visible_message("<span class='chaosverybad'>[H] looks like they're about to die!</span>", "<span class='chaosverybad'>HEARTUS DELETUS!</span>")
+			var/obj/item/organ/internal/heart/target_heart = H.get_int_organ(/obj/item/organ/internal/heart)
+			if(target_heart)
+				target_heart.remove(H)
+				qdel(target_heart)
+		if("gibbed")
+			H.visible_message("<span class='chaosverybad'>[H] falls into gibs!</span>", "<span class='chaosverybad'>Oof!</span>")
+			H.gib()
+		if("cluwned")
+			H.visible_message("<span class='chaosverybad'>[H] turns into a cluwne!</span>", "<span class='chaosverybad'>Oh no.</span>")
+			H.makeCluwne()
+		if("spaced")
+			for(var/obj/item/I in H)
+				H.unEquip(I, TRUE)
+			var/turf/T = safepick(get_area_turfs(/area/space/nearstation)) //Send in space next to the station
+			if(!T) //Shouldn't happen but just in case
+				T = safepick(get_area_turfs(/area/space))
+			if(!T) //What do you mean there's no space? Okay well just die then
+				H.visible_message("<span class='chaosverybad'>[H] drops dead!</span>", "<span class='chaosverybad'>Game over!</span>")
+				H.death(FALSE)
+			else
+				H.visible_message("<span class='chaosverybad'>[H] disappears!</span>", "<span class='chaosverybad'>COLD! CAN'T BREATH!</span>")
+				do_teleport(H, T)
+		if("decapitated")
+			H.visible_message("<span class='chaosverybad'>[H]'s head goes flying!'</span>", \
+				"<span class='chaosverybad'>You watch the floor fly to your face as you rapidly lose consciousness...</span>")
+			var/obj/item/organ/external/affected = target.get_organ("head")
+			var/atom/movable/A = affected.droplimb(1, DROPLIMB_SHARP)
+			INVOKE_ASYNC(A, TYPE_PROC_REF(/atom/movable, throw_at), pick(oview(7, get_turf(src))), 10, 1)
+		if("banned")
+			H.visible_message("<span class='chaosverybad'>[H] gets <span class='adminhelp'>BWOINKED</span> out of existence!</span>", \
+				"<span class='chaosverybad'>You get <span class='adminhelp'>BWOINKED</span> out of existence!</span>")
+			playsound(H, 'sound/effects/adminhelp.ogg', 100, FALSE)
+			qdel(H)
+		if("exploded")
+			H.visible_message("<span class='chaosverybad'>[H] explodes!</span>", "<span class='chaosverybad'>Boom!</span>")
+			explosion(get_turf(H), 1, 1, 1, cause = "staff of chaos lethal explosion effect")
+		if("cheese morphed")
+			H.visible_message("<span class='chaosverybad'>[H] transforms into cheese!</span>", "<span class='chaosverybad'>You've been transformed into cheese!</span>")
+			new /obj/item/reagent_containers/food/snacks/cheesewedge(get_turf(H))
+			qdel(H)
+		if("supermattered")
+			var/obj/machinery/atmospherics/supermatter_crystal/supercrystal = locate(/obj/machinery/atmospherics/supermatter_crystal)
```

Comment:
```suggestion
			var/obj/machinery/atmospherics/supermatter_crystal/supercrystal = GLOB.main_supermatter_engine
```
might as well just do this since this is going to be the outcome 99% of the time and is infinitely faster. Either this or tracking all SMs in a global list and then just doing pick() on that